server/rpc.go

```diff
@@ -50,6 +51,80 @@ type Broadcaster interface {
 	Sign([]byte) ([]byte, error)
 }
 
+type BroadcastSessionsManager struct {
+	broadcastSessions []*BroadcastSession
+	sessLock          *sync.Mutex
+}
+
+// func (bsm *BroadcastSessionsManager) selectFromList(node *core.LivepeerNode, pl core.PlaylistManager) *BroadcastSession {
+func (bsm *BroadcastSessionsManager) selectFromList() *BroadcastSession {
+	// bsm.checkStatusOfList(numOrchs, node, pl)
+	bsm.checkStatusOfList()
+
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+
+	sess, sessions := bsm.broadcastSessions[0], bsm.broadcastSessions[1:]
+	bsm.broadcastSessions = sessions
+
+	return sess
+}
+
+func (bsm *BroadcastSessionsManager) removeFromList(session *BroadcastSession) {
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+	var sessions []*BroadcastSession
+	for i, s := range bsm.broadcastSessions {
+		if s == session {
+			sessions = append(bsm.broadcastSessions[:i], bsm.broadcastSessions[i+1:]...)
+			break
+		}
+	}
+	bsm.broadcastSessions = sessions
+}
+
+func (bsm *BroadcastSessionsManager) addToList(sess *BroadcastSession) {
+	bsm.sessLock.Lock()
+	bsm.broadcastSessions = append(bsm.broadcastSessions, sess)
+	bsm.sessLock.Unlock()
+}
+
+// func (bsm *BroadcastSessionsManager) checkStatusOfList(numOrchs int, node *core.LivepeerNode, pl core.PlaylistManager) {
+func (bsm *BroadcastSessionsManager) checkStatusOfList() {
+	var newBroadcastSessions []*BroadcastSession
+	defaultNumOrchs := int(HTTPTimeout / SegLen)
```

Comment:
maybe a note this is the maximum number of inflight sessions.

Since `defaultNumOrchs`, is probably better described as "the minimum number of orchestrators we need before we'll try to refresh the orchestrator list", maybe call it `minOrchs` or something